File: Code/Chapter 8/VB.NET/Exercise 8.8/Form1.vb
Imports CreateDataSet
Imports System.Data

Public Class Form1
   Private dataFilePath As String = "..\..\..\..\Data.Xml"
   Private CustomersTable As DataTable
   Private CustomersView As DataView

   Private Sub btnLoad_Click( _
     ByVal sender As System.Object, ByVal e As System.EventArgs) _
     Handles btnLoad.Click
      CustomersTable = _
         CreateDataSet.DataSetFiller.FillDataset(dataFilePath).Tables("Customers")
      CustomersView = New DataView(CustomersTable)
      dgView.DataSource = CustomersView
   End Sub
   Private Sub btnSort_Click( _
     ByVal sender As System.Object, ByVal e As System.EventArgs) _
     Handles btnSort.Click
      CustomersView.Sort = "FirstName ASC"
   End Sub
   Private Sub btnFilter_Click( _
     ByVal sender As System.Object, ByVal e As System.EventArgs) _
     Handles btnFilter.Click
      CustomersView.RowFilter = "LastName like '%OfJungle'"
   End Sub
   Private Sub btnFindRows_Click( _
     ByVal sender As System.Object, ByVal e As System.EventArgs) _
     Handles btnFindRows.Click
      Dim drvs() As DataRowView = CustomersView.FindRows("Tarzan")
      Dim drv As DataRowView
      For Each drv In drvs
         MessageBox.Show(drv.Row("FirstName") & " " & drv.Row("LastName"), "Selected Item")
      Next
   End Sub
End Class

File: Code/Chapter 8/C#/Exercise 8.8/Form1.cs
#region Using directives

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Windows.Forms;
#endregion

namespace Exercise_8_8
{
   partial class Form1 : Form
   {
      private static string dataFilePath = @"..\..\..\..\..\Data.Xml";
      private DataTable CustomersTable;
      private DataView CustomersView;

      public Form1()
      {
         InitializeComponent();
      }

      private void btnLoad_Click(object sender, EventArgs e)
      {
         CustomersTable = CreateDataSet.DataSetFiller.FillDataset(dataFilePath).Tables["Customers"];
         CustomersView = new DataView(CustomersTable);
         dgView.DataSource = CustomersView;
      }

      private void btnSort_Click(object sender, EventArgs e)
      {
         CustomersView.Sort = "FirstName ASC";
      }

      private void btnFilter_Click(object sender, EventArgs e)
      {
         CustomersView.RowFilter = "LastName like '%OfJungle'" ;
      }

      private void btnFindRows_Click(object sender, EventArgs e)
      {
         DataRowView[] drvs = CustomersView.FindRows("Tarzan");
         foreach (DataRowView drv in drvs)
         {
            MessageBox.Show(drv.Row["FirstName"] + " " + drv.Row["LastName"], "Selected Item");
         }
      }
    }
}